src/built-in-macros.ls

```diff
@@ -244,20 +244,49 @@ contents =
 
   \. : do
 
-    is-computed-property = (ast-node) ->
-      switch ast-node.type
-      | \Identifier => false
-      | otherwise => true
+    # Terminology:
+    #
+    #  - a _computed_ member expression involves square brackets, like `a[b]`.
+    #  - a _non-computed_ member expression involves a dot like `a.b`.
+
+    # Assumptions:
+    #
+    # - The estree spec requires a non-computed member expression's property to
+    #   be an Identifer node.
+    # - There's no reason we'd ever want to write `a["b"]`, as that is
+    #   completely equivalent to `a.b` and easier to read.
+
+    make-member-expression = ({compile}:env, object, property) ->
+
+      var computed
+      var property-compiled
+
+      # Compile the property to an estree node, so we can decide what to do
+      # with it
+      property-as-estree = compile property
+
+      # Ensure that a string literal is converted to an identifer and compiled
+      # into a non-computed member expression
+      if property-as-estree.type is \Literal
+      and typeof property-as-estree.value is \string
```

Comment:
You need to ensure here that the value of the literal is valid in a dot expression.  It would suck if an author wrote `(. a "foo-bar")` (assuming it would desugar to the valid expressions `a["foo-bar"]`) and got an error instead.  (As currently written, they'd have to work around it with something like `(. a (+ "foo-bar"))` to force it down the "computed" path.)

Per the Gitter chat, while reserved keywords are now valid in dot expressions, it's probably also good to exclude them here, just to maximize compatibility with older browsers.

See <https://mathiasbynens.be/notes/javascript-properties> for details on what's valid or not.